flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -124,11 +126,33 @@ public void initializeState(FunctionInitializationContext context) throws Except
     if (context.isRestored()) {
       LOG.info("Restoring state for the {}.", getClass().getSimpleName());
       lastSnapshotId = lastSnapshotIdState.get().iterator().next();
+    } else if (scanContext.startTag() != null) {
+      Snapshot currentSnapshot =
+          scanContext.branch() == null
+              ? table.currentSnapshot()
+              : table.snapshot(scanContext.branch());
+      Preconditions.checkNotNull(
+          currentSnapshot,
+          "Don't have any available snapshot for branch " + scanContext.branch() + " in table.");
```

Comment:
I think it'll help if at some point (probably options/conf default) we set the default branch to main and always assume a branch is set in this code. Then we can just do table.snapshot(branch).

Nit on log message:

"No snapshots on branch %s in table %s"